Entity/Components/HealthComponent.gd

```diff
@@ -41,18 +39,67 @@ func modify_health(amount: float, caster: Entity, is_healing: bool = false) -> v
 	if caster == target: 
 		caster = null
 
-	# apply modification to our health
-	if is_healing:
-		new_health = clampf(current_health + amount, 0, max_health)
-	else:
-		new_health = clampf(current_health - amount, 0, max_health)
-		
+	
+	var leftover_damage: float
+	leftover_damage = amount
+	
+	leftover_damage = block_damage(leftover_damage)
+	_health_damage(leftover_damage)
+
+## removes health without considering block [br]
+## primarily a helper function, but could be used in future for block ignoring attacks[br]
+func _health_damage(damage : int) -> void:
+	var new_health : int
+	new_health = current_health
+	
+	new_health = clampf(current_health - damage, 0, max_health)
 	_set_health(new_health)
 
+## adds health to the unit[br]
+func heal(amount : int,  _caster : Entity) -> void:
+	var new_health : int
+	new_health = current_health
+	
+	new_health = clampf(current_health + amount, 0, max_health)
+	_set_health(new_health)
 
 ## Set the health of the entity [br]
 func _set_health(new_health: float) -> void:
 	if (new_health == current_health):
 		return
 	current_health = new_health
 	on_health_changed.emit(current_health)
+
+## Removes block from the entity[br]
+func block_damage(damage: int) -> int:
+	var new_block : int
+	new_block = current_block
+	
+	var leftover_damage : int
+	leftover_damage = max( damage - current_block, 0)
+	
+	new_block = max(new_block - damage, 0)
```

Comment:
`new_block = max(current_block - damage, 0)`
i know that technically `new_block = current_block` but I find it clearer that way, as the block as not been changed yet